common/lib/xmodule/xmodule/library_content_module.py

```diff
@@ -163,6 +164,13 @@ class LibraryContentBlock(
         default=[],
         scope=Scope.user_state,
     )
+    display_reset_button = Boolean(
```

Comment:
```suggestion
    # This cannot be called `show_reset_button`, because children blocks inherit this as a default value.
    allow_resetting_children = Boolean(
```

How about this? `show_reset_button` was, after all, a pretty inaccurate name - we are returning HTTP 400 in `reset_selected_children` if it's not enabled.

I suspect that this bug could be related to the [InheritanceMixin](https://github.com/edx/edx-platform/blob/898684287a71d4253c4c21819caf9e23ce7259f9/common/lib/xmodule/xmodule/modulestore/inheritance.py#L198-L207), which allows overriding default values for the `show_reset_button` per course via Advanced Settings. I haven't verified this, though - it was only my initial thought.

Reply:
Awesome